src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -875,6 +1042,16 @@ int dhcp_lease_parse_options(uint8_t code, uint8_t len, const void *option, void
                 break;
         }
 
+        case SD_DHCP_OPTION_V4_DNR:
+                //FIXME: remove matching entries from DNS so they aren't duplicated?
```

Comment:
that certainly might make sense, but this should be done after all options are processed, where we basically clean up what we learnt.

Reply:
I've decided not to implement this for now, so we will keep all DNS/DNR servers we have learned.